packages/react-instantsearch-hooks-dom/src/widgets/__tests__/RefinementList.test.tsx

```diff
@@ -0,0 +1,3462 @@
+
+      await wait(0);
+
+      expect(
+        Array.from(container.querySelectorAll('.ais-RefinementList-count')).map(
+          (item) => item.textContent
+        )
+      ).toEqual([
+        '746',
+        '633',
+        '591',
+        '530',
+        '442',
+        '394',
+        '350',
+        '320',
+        '318',
+        '298',
+      ]);
+    });
+
+    test('sorts the items by refinement state', async () => {
+      const client = createSearchClient({ search });
+      const { container } = render(
+        <InstantSearchHooksTestWrapper searchClient={client}>
+          <RefinementList attribute="brand" sortBy={['isRefined']} />
+        </InstantSearchHooksTestWrapper>
+      );
+
+      await wait(0);
+
+      expect(
+        Array.from(
+          container.querySelectorAll('.ais-RefinementList-labelText')
+        ).map((item) => item.textContent)
+      ).toEqual([
+        'Insignia™',
+        'Samsung',
+        'Metra',
+        'HP',
+        'Apple',
+        'GE',
+        'Sony',
+        'Incipio',
+        'KitchenAid',
+        'Whirlpool',
+      ]);
+
+      const [checkbox1, checkbox2] = Array.from(
+        container.querySelectorAll(
+          '.ais-RefinementList-checkbox'
+        ) as NodeListOf<HTMLInputElement>
+      ).slice(-2);
+
+      userEvent.click(checkbox1);
+      userEvent.click(checkbox2);
+
+      await wait(0);
+
+      expect(
+        Array.from(
+          container.querySelectorAll('.ais-RefinementList-labelText')
+        ).map((item) => item.textContent)
+      ).toEqual([
+        'KitchenAid',
+        'Whirlpool',
+        'Insignia™',
+        'Samsung',
+        'Metra',
+        'HP',
+        'Apple',
+        'GE',
+        'Sony',
+        'Incipio',
+      ]);
+    });
+
+    test('sorts the items using a sorting function', async () => {
+      const client = createSearchClient({ search });
+      const { container } = render(
+        <InstantSearchHooksTestWrapper searchClient={client}>
+          <RefinementList
+            attribute="brand"
+            sortBy={(a, b) => a.name.length - b.name.length}
+          />
+        </InstantSearchHooksTestWrapper>
+      );
+
+      await wait(0);
+
+      expect(
+        Array.from(
+          container.querySelectorAll('.ais-RefinementList-labelText')
+        ).map((item) => item.textContent)
+      ).toEqual([
+        'HP',
+        'GE',
+        'LG',
+        'Sony',
+        'Dell',
+        'Metra',
+        'Apple',
+        'Canon',
+        'Speck',
+        'Epson',
+      ]);
+    });
+  });
+
+  describe('searching', () => {
+    test('displays a search box', async () => {
+      const client = createSearchClient({ search, searchForFacetValues });
+      const { container } = render(
+        <InstantSearchHooksTestWrapper searchClient={client}>
+          <RefinementList
+            attribute="brand"
+            searchable={true}
+            searchablePlaceholder="Search brands"
+          />
```

Comment:
We may want to use `limit` to not bloat the whole file with snapshots.

Reply:
I went with more focused tests on just the list of returned facets (see [445448c](https://github.com/algolia/react-instantsearch/pull/3381/commits/445448cb0cadfd537e1e71a6f7074db1e9789439))